### ISSUE TITLE
ENH: Geometry native type to Iceberg connector

### ISSUE THREAD

**Issue by kaushiksrini** [trinodb/trino#23512]
## Description

I recently saw that the Geometry type was accepted [into the parquet format](https://github.com/apache/parquet-format/pull/240) (but not yet merged). We are looking forward for these changes to be merged and have the type supported by the native parquet format. Following this, there seems to be interest in adding a [geometry native type to iceberg](https://github.com/apache/iceberg/issues/10260). After this, it would be simple for us to use the underlying geometry type to perform geospatial operations on the data without having to call `ST_GeometryFromText`, as this operation is [quite expensive](https://github.com/prestodb/presto/issues/9890#:~:text=comments-,ParseWktCost,per%20point,-NewPointCost). I am looking to add this geometry feature to the iceberg connector, once the Iceberg changes have been implemented. I am creating this issue as a placeholder for the work. Let me know if there is already work being done on this, otherwise happy to work on it. Thanks!

**Comment by MrPowers** [comment]
The Iceberg geometry PR was merged, [see here](https://github.com/apache/iceberg/pull/10981).

Let me know if you're willing to work on this, and I'd be happy to help point you in the right direction!

**Comment by kaushiksrini** [comment]
@MrPowers Yep, I can begin working on this.

**Comment by ebyhr** [comment]
@kaushiksrini I don't think you can start this work because the connector doesn't support V3 spec yet.

**Comment by PeterAustinMoore** [comment]
I think since we are using the 1.9.1 release of the org.apache.iceberg libraries, the geotypes should be available now, yes?

**Comment by ebyhr** [comment]
@PeterAustinMoore No, please see the above message.

**Comment by kaushiksrini** [comment]
@ebyhr - given that https://github.com/apache/iceberg/pull/12667 is in the final stages, how far is the connector in supporting the V3 spec?

**Comment by dain** [comment]
I have a PR for Iceberg V3 geo types here.  Note this only implements the types, and does not support stats or any kind of pushdown
